user/private-dependencies-gl.md

```diff
@@ -0,0 +1,196 @@
+>   - echo -e "machine gitlab.com\n  login ci-user\n  password $CI_USER_PASSWORD" >~/.netrc
+>   - git submodule update --init --recursive
+> ```
+> {: data-file=".travis.yml"}
+
+## API Token
+
+Assumptions:
+
+- The repository you are running the builds for is called "myorg/main" and depends on "myorg/lib1" and "myorg/lib2".
+- You know the credentials for a user account that has at least read access to all three repositories.
+
+This approach works just like the [password](#password) approach outlined above, except instead of the username/password pair,
+you use a GitLab API token.
+
+Under the GitLab account settings for the user you want to use, navigate to
+
+and then generate a "Personal access tokens".
+
+Make sure the token has the "repo" scope.
+
+Your `~/.netrc` should look like this:
+
+```
+machine gitlab.com
+  login the-generated-token
+```
+
+You can also use it in URLs directly: `https://the-generated-token@gitlab.com/myorg/lib1.git`.
+
+Use the `encrypt` command to add the token to your `.travis.yml`.
+
+```bash
+$ travis env set CI_USER_TOKEN the-generated-token --private -r myorg/main
+```
+
+You can then have Travis CI write to the `~/.netrc` on every build.
+
+```yaml
+before_install:
+- echo -e "machine gitlab.com\n  login $CI_USER_TOKEN" > ~/.netrc
+```
+{: data-file=".travis.yml"}
+
+It is also possible to inject the token into URLs, for instance, in a Gemfile, it would look like this:
+
+```ruby
+source 'https://rubygems.org'
+gemspec
+
+if ENV['CI']
+  # use HTTPS with token on Travis CI
+  git_source :gitlab do |repo_name|
+    repo_name = "#{repo_name}/#{repo_name}" unless repo_name.include?("/")
+    "https://#{ENV.fetch("CI_USER_TOKEN")}@gitlab.com/#{repo_name}.git"
+  end
+end
+
+gem 'lib1', gitlab: "myorg/lib1"
+gem 'lib2', gitlab: "myorg/lib2"
+```
+
+> In case of private git submodules, be aware that the `git submodule
+> update --init --recursive` command runs before the `~/.netrc` credentials
+> are updated. If you are writing credentials to `~/.netrc`, disable the automatic loading of
+> submodules, update the credentials and add an explicit step to update the submodules:
+>
+> ```yaml
+> git:
+>   submodules: false
+> before_install:
+>   - echo -e "\n\nmachine gitlab.com\n  $CI_TOKEN\n" >~/.netrc
+>   - git submodule update --init --recursive
+> ```
+> {: data-file=".travis.yml"}
+
+## Dedicated User Account
+
+As mentioned a few times, it might make sense to create a dedicated CI user for the following reasons:
+
+- The CI user will only have access to the repositories you want it to have access to.
+- You can limit the access to read access.
+- Less risk when it comes to leaking keys or credentials.
+- The CI user will not leave the organization for non-technical reasons and accidentally break all your builds.
+
+In order to do so, you need to register on GitLab as if you would be signing up for a normal user.
+Registering users cannot be automated, since that would violate the GitLab Terms of Service.
```

Comment:
@weban, please take a closer look at this section to make sure all the keys and tokens are correct for GL